### PR TITLE
Settings and Logging

### PR DESCRIPTION
**Pull request by veda-consulting**
Changes to remove the .env file and instead use a settings page in CiviCRM
Allow users to pick the custom field they wish to use in the settings page and not have to know the ID
Added logging into the CiviRules actions so that, if using the CiviRules Logging extension, you can see wha the issues are
Updated the README to reflect changes

### CODE REVIEW

**Comment by rupertraphael** [comment]
Thank so much for this, @veda-consulting! We're still deciding where to test it (but it looks like it works as intended already). Either on the client site or in a fresh install each of which has its hurdles - the former, we just need to talk to the client while with the latter, subscribe back to zoom webinars. I'll give you an update when we get to test it already.

**Comment by veda-consulting** [comment]
@rupertraphael no probs, we're running our clients off of our fork

I'll be doing a bit of a video and promotion on civicrm.org so watch this space

**Comment by rupertraphael** [comment]
@veda-consulting After a long hiatus, I was finally able to find time to test this. Again, awesome work by you guys! I just have a few suggestions (including the one below).

Suggestion: I think there would be use-cases wherein users only want to work with just meetings or just webinars. What do you think about making the _Custom Field_ and _Custom Field for Meeting_ settings fields optional?

File: CRM/CivirulesActions/Participant/AddToZoom.php [hunk 1]
```diff
@@ -21,7 +21,13 @@ public function processAction(CRM_Civirules_TriggerData_TriggerData $triggerData
 
 	  $participant = $this->getContactData($contactId);
 
-	  $this->addParticipant($participant, $webinar, $triggerData);
+	  $meeting = $this->getMeetingID($event['id']);
+
+	  if(!empty($meeting)){
+	  	$this->addParticipant($participant, $meeting, $triggerData, 'Meeting');
+	  } elseif (!empty($webinar)) {
+	  	$this->addParticipant($participant, $webinar, $triggerData, 'Webminar');
```
Comment (rupertraphael) [inline CRM/CivirulesActions/Participant/AddToZoom.php:24]: Please fix typo: 'Webminar' to 'Webinar'

File: CRM/CivirulesActions/Participant/AddToZoom.php [hunk 2]
```diff
@@ -79,14 +109,19 @@ private function getContactData($id) {
...
 	 * @param array $participant participant data where email, first_name, and last_name are required
-	 * @param int $webinar id of an existing Zoom webinar
+	 * @param int $entityID id of an existing Zoom webinar/meeting
+	 * @param string $entity 'Meeting' or 'Webminar'
 	 */
-	private function addParticipant($participant, $webinar, $triggerData) {
+	private function addParticipant($participant, $entityID, $triggerData, $entity) {
 		$settings = CRM_NcnCiviZoom_Utils::getZoomSettings();
-		$url = $settings['base_url'] . "/webinars/$webinar/registrants";
+		if($entity == 'Webminar'){
```
Comment (rupertraphael) [inline CRM/CivirulesActions/Participant/AddToZoom.php:123]: Please fix typo: 'Webminar' to 'Webinar'

File: templates/CRM/NcnCiviZoom/Form/Settings.tpl [hunk 3]
```diff
@@ -11,6 +11,11 @@
       </br>
       {ts}Select the event custom field which holds the Zoom Webminar ID{/ts}
```
Comment (rupertraphael) [inline templates/CRM/NcnCiviZoom/Form/Settings.tpl:73]: Please fix typo: 'Webminar' to 'Webinar'

File: CRM/NcnCiviZoom/Form/Settings.php [hunk 4]
```diff
@@ -20,36 +23,92 @@ public function buildQuickForm() {
...
       'size' => 48,
     ), TRUE);
-    $this->add('text', 'custom_field_id', ts('Custom Field Id'), array(
-      'size' => 48,
-    ), TRUE);
-    $this->addButtons([
+    $this->add(
+      'select',
+      'custom_field_id',
+      'Custom Field',
```
Comment (rupertraphael) [inline CRM/NcnCiviZoom/Form/Settings.php]: Suggestion: can we turn this label into 'Custom Field for Webinar'?

**Comment by rupertraphael** [comment]
Closing PR this as I haven't had the time to review the code and maintain this project. Thank you @veda-consulting for all your hardwork.

For those who might bump into this repo, please go to https://github.com/veda-consulting-company/ncn-civi-zoom

**Pull request closed without merging**